ehcache-impl/src/test/java/org/ehcache/impl/persistence/DefaultLocalPersistenceServiceTest.java

```diff
@@ -140,4 +142,38 @@ public void testExclusiveLock() throws IOException {
     RuntimeException thrown = assertThrows(RuntimeException.class, () -> service2.start(null));
     assertThat(thrown, hasProperty("message", is("Persistence directory already locked by this process: " + testFolder.getAbsolutePath())));
   }
+
+  @Test
+  public void testServiceStoppedStatusWhenStoppedCleanly() throws IOException {
+    File f = folder.newFolder("testServiceStoppedStatusWhenStoppedCleanly");
+    final DefaultLocalPersistenceService service = new DefaultLocalPersistenceService(new DefaultPersistenceConfiguration(f));
+    service.start(null);
+    service.stop();
+    service.start(null);
+    assertTrue(service.isClean());
+    service.stop();
+  }
+
+  @Test
+  public void testServiceStoppedStatusWhenStoppedUnexpectedly() throws IOException {
+    // Service stopped unexpectedly means directory exists with some data but without .clean file.
+    File f = folder.newFolder("testServiceStoppedStatusWhenStoppedUnexpectedly");
+    final DefaultLocalPersistenceService service = new DefaultLocalPersistenceService(new DefaultPersistenceConfiguration(f));
+    new File(f.getAbsolutePath()+"\\dummy.txt").createNewFile();
```

Comment:
This doesn't work correctly on any platform except Windows. Outside of Windows you'll just get a file named `<blahblahblah>\dummy.txt` because `\` is not the right separator. You should use the `File(File, String)` constructor instead.